### DISCUSSION TITLE
Porting GSM Modem Example to STM32 Nucleo-U5A5ZI-Q with SIM7600

### DISCUSSION THREAD

**Discussion by srikanthpalvai** [zephyrproject-rtos/zephyr#72194] [General]
Hello Zephyr community,

I am currently working on a project where I need to integrate a GSM modem with the Zephyr RTOS. I found the GSM modem example in the Zephyr repository, which is built with the B-U585I-IOT02A board and BG96 modem.

However, I need to port this example to work with the Nucleo-U5A5ZI-Q board, which uses the SIM7600 modem. I'm relatively new to Zephyr development, so I would appreciate some guidance on how to proceed with this porting effort.

Specifically, I would like to know:

What are the key differences between the BG96 and SIM7600 modems in terms of hardware interfaces and communication protocols?
Are there any existing Zephyr drivers or libraries for the SIM7600 modem that I can leverage?
What steps do I need to take to adapt the existing GSM modem example code to work with the SIM7600 modem and the Nucleo-U5A5ZI-Q board?
Are there any potential challenges or considerations I should be aware of during the porting process?
Any insights, suggestions, or resources you can provide would be greatly appreciated.

Thank you in advance for your help!

Best regards,

**Comment by jukkar** [comment]
You did not mention what Zephyr version you are using, there is the "newish" modem subsystem you should use in your porting efforts. For a reference the `samples/net/cellular_modem` is using the new modem subsystem, the `samples/net/gsm_modem` is the legacy generic GSM modem driver you should not use (it will be removed eventually).
For the difference between BG96 and SIM7600, you could ask in #modem channel in Zephyr Discord.

**Reply by srikanthpalvai** [reply]
Thanks for the quick response 
zephyr-sdk-0.16.5-1
*** Booting Zephyr OS build v3.6.0-2753-g45205f865e18 ***

**Comment by srikanthpalvai** [comment]
@jukkar 
I was able to communicate with my sim7080 modem 
Is there any example where I can extract the main logic of PPP with CMUX.
instead of modem_chat scripts,  I will fire my custom sequence at commands.

**Comment by davidalo** [comment]
Hello,

Following this. I am trying to integrate SIM900, running Zephyr v3.7 (@srikanthpalvai there are big differences from v3.6.0 to v3.7.0 so you may want to upgrade). However, I am struggling to communicate over DLC 2 for AT commands, I can not get any answer from the modem, only echo.

**Comment by srikanthpalvai** [comment]
@davidalo 
Suggestion i got from the zephyr team is try with the cellular modem.
i think it better to try with cellular modem 
here you can choose your hardware though the the conf file 
https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/net/cellular_modem

**Reply by davidalo** [reply]
@srikanthpalvai I'm working with cellular modem, but I'm still facing issues. I will post a different discussion with more details and link you there.

**Reply by davidalo** [reply]
Here you have a better description of my issue: https://github.com/zephyrproject-rtos/zephyr/discussions/76894
